engine/src/main/java/org/terasology/engine/particles/components/ParticleEmitterComponent.java

```diff
@@ -110,4 +110,26 @@
      * Allows checking only some particles each update since it's a heavy operation.
      */
     public int collisionUpdateIteration;
+
+    @Override
+    public void copy(ParticleEmitterComponent other) {
+        this.maxParticles = other.maxParticles;
+        this.particleCollision = other.particleCollision;
+        this.spawnRateMax = other.spawnRateMax;
+        this.spawnRateMin = other.spawnRateMin;
+        this.enabled = other.enabled;
+        this.lifeTime = other.lifeTime;
+        this.particleSpawnsLeft = other.particleSpawnsLeft;
+        this.destroyEntityWhenDead = other.destroyEntityWhenDead;
+        this.ownerEntity = other.ownerEntity;
+        this.particlePool = other.particlePool;
+        this.generatorFunctionMap.clear();
+        this.generatorFunctionMap.putAll(other.generatorFunctionMap);
```

Comment:
Is just putting all values here enough, or do we need to create deep copies here? This again is a direct reference from one component to another, which we should try to avoid.. 

Same holds for `affectorFunctionMap`.